File: task2/py/pages/example_page.py
from playwright.sync_api import Page # type: ignore

class ExamplePage:
    def __init__(self, page: Page):
        self.page = page
        self.more_info_link = "a:has-text('More information')"

    def load(self):
        self.page.goto("https://example.com")
    
    def get_title(self) -> str:
        return self.page.title()
    
    def click_more_info(self):
        self.page.click(self.more_info_link)
    
    def get_current_url(self) -> str:
        return self.page.url

File: task2/ts/pages/example.page.ts
import { Page, expect } from '@playwright/test';

export class ExamplePage {
  private readonly page: Page;

  constructor(page: Page) {
    this.page = page;
  }

  async navigate() {
    await this.page.goto('https://example.com');
  }

  async verifyTitle() {
    await expect(this.page).toHaveTitle(/Example/);
  }

  async clickMoreInformation() {
    await this.page.click('css=a:has-text("More information")');
  }
}
